thirdparty/DiskANN/src/pq_flash_index.cpp

```diff
@@ -842,12 +843,33 @@ namespace diskann {
     const T *    query = data.scratch.aligned_query_T;
     const float *query_float = data.scratch.aligned_query_float;
 
-    for (uint32_t i = 0; i < this->data_dim; i++) {
+    auto q_dim = this->data_dim;
+    if(metric== diskann::Metric::INNER_PRODUCT && !smaple){ // query_dim need specify when using IP
+      q_dim--;
+    }
+    for (uint32_t i = 0; i < q_dim; i++) {
       data.scratch.aligned_query_float[i] = query1[i];
       data.scratch.aligned_query_T[i] = query1[i];
       query_norm += query1[i] * query1[i];
     }
 
+    if(query_norm <= 0 && metric == diskann::Metric::INNER_PRODUCT){
```

Comment:
To my understanding, if query_norm == 0, DiskANN will crash right? And did we meet this situation before?

Reply:
No, it wouldn't crash, because query1's mem allocated at heap. if access the out-of-bounds mem, it would get a random value.

Reply:
> To my understanding, if query_norm == 0, DiskANN will crash right? And did we meet this situation before?

No, it wouldn't crash, because query1's mem allocated at heap. if access the out-of-bounds mem, it would get a random value.

Reply:
Interesting, did you see this? the number should be extremely big and the normalization will be super weird.